From 773f463460f8fbb5656492c2cd419a2d2a70b006 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Sun, 14 Aug 2016 10:46:31 +0300
Subject: [PATCH 0402/1345] pp3: fw and drv version print stile adjust

commit  a717d3d26404a11e893df2f2c6fe8291168b4651 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Although the dmesg"
Reconsider sprintf/pr_info with carriage-return.
Problem is: although the "dmesg" print is correct
the real-time print has bad stile print
    FW version:     NSS:16.08.0[13250.770044]
Adjust also the  for 3/4 digit printing for the
    "Driver version"

Change-Id: I3fe8f171fb8881da9feb6e04346e764b118ec81b
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31909
Reviewed-by: Dmitri Epshtein <dima@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yelena Krivosheev <yelena@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/pp3/platform/mv_pp3.c |   15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/marvell/pp3/platform/mv_pp3.c b/drivers/net/ethernet/marvell/pp3/platform/mv_pp3.c
index b2b7051..86d55ba 100644
--- a/drivers/net/ethernet/marvell/pp3/platform/mv_pp3.c
+++ b/drivers/net/ethernet/marvell/pp3/platform/mv_pp3.c
@@ -621,6 +621,7 @@ int mv_pp3_shared_start(struct mv_pp3 *priv)
 	struct mv_pp3_version fw_ver, *drv_ver;
 	char *version_name;
 	int ver_name_size;
+	char buf[80];
 
 	if (priv->initialized)
 		return 0;
@@ -684,17 +685,19 @@ int mv_pp3_shared_start(struct mv_pp3 *priv)
 	drv_ver = mv_pp3_get_driver_version();
 	memcpy(version_name, drv_ver->name, ver_name_size);
 	pr_info("\n");
-	pr_info("Driver version: %s:%02d.%02d.%d",
+	sprintf(buf, "Driver version: %s:%02d.%02d.%d",
 		version_name, drv_ver->major_x, drv_ver->minor_y, drv_ver->local_z);
 	if (drv_ver->debug_d)
-		pr_cont(".%d\n", drv_ver->debug_d);
-
+		pr_info("%s.%d\n", buf, drv_ver->debug_d);
+	else
+		pr_info("%s\n", buf);
 	memcpy(version_name, fw_ver.name, ver_name_size);
-	pr_info("FW version:     %s:%02d.%02d.%d",
+	sprintf(buf, "FW version:     %s:%02d.%02d.%d",
 		version_name, fw_ver.major_x, fw_ver.minor_y, fw_ver.local_z);
 	if (fw_ver.debug_d)
-		pr_cont(".%d\n", fw_ver.debug_d);
-
+		pr_info("%s.%d\n", buf, fw_ver.debug_d);
+	else
+		pr_info("%s\n", buf);
 	pr_info("\n");
 	kfree(version_name);
 
-- 
1.7.9.5

